pkg/ebpf/map_cleaner.go

```diff
@@ -8,70 +8,67 @@
 package ebpf
 
 import (
-	"bytes"
-	"encoding"
-	"encoding/binary"
-	"errors"
-	"fmt"
-	"reflect"
 	"sync"
 	"time"
 	"unsafe"
 
 	cebpf "github.com/cilium/ebpf"
 
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
-	"github.com/DataDog/datadog-agent/pkg/util/native"
 )
 
 // MapCleaner is responsible for periodically sweeping an eBPF map
 // and deleting entries that satisfy a certain predicate function supplied by the user
-type MapCleaner struct {
-	emap *cebpf.Map
-	key  interface{}
-	val  interface{}
-	once sync.Once
+type MapCleaner[K any, V any] struct {
+	emap         *cebpf.Map
+	keyBatch     []K
+	valuesBatch  []V
+	keysToDelete []K
```

Comment:
One concern with keeping the slice here, is this slice will only ever grow, so a burst of cleaning will cause allocations that will never be recovered.

Reply:
re: `keyBatch` and `valuesBatch` - those cannot grow more than the initial creation, so those are limited to the `batchSize`.

re: keysToDelete - I added a call to shrink its size to batchLength at the end of the operaitons.